cirq/__init__.py

```diff
@@ -537,6 +537,9 @@
     PauliSumCollector,
     Sampler,
     Collector,
+    InitObsSetting,
```

Comment:
I think this should be the "last Observable Measurement PR"'s job to expose everything on the top level. It is typical to implement large features progressively and "enable them" for users when they are finished. I think about adding to `cirq/__init__.py` as exposing it as a flag / turning on a feature flag. While OM is in progress these should be "hidden".  In case we don't make it across the finish line until v0.10.0 (I hope we will though!) - we'll just have a bunch of unusable classes in the cirq namespace.

Reply:
Alternatively we could mark these classes as WIP...

Reply:
This is a good idea, but it gets complicated with JSON serialization and REPRs. I've split out the json serialization (and adding to api.rst) to be applied at the end during the flag-flip. A downside is we can't test the json serialization incrementally (other than to point to the earlier version of this PR that had it and passed tests)

Reply:
Great point and thank you for "hiding" the classes for now! 

It is actually not that bad to make JSON serialization work and we should do it to keep it tested incrementally!  

1. You could add `    result.update(_get_all_public_classes(cirq.work))` here in the json_serialization_test.py for coverage: 
https://github.com/quantumlib/Cirq/blob/ca307cd7ffdd2e8659c84c484788c8b5d15cc09d/cirq/protocols/json_serialization_test.py#L345-L353 with the comment that 

Then everything else is pretty much the usual: 

1. Recreate both the repr (this should have cirq.work.InitObsSetting and cirq.work.MeasurementSpec) and json files as usual.
2. register the types in the `_ResolverCache` as usual but point to the `cirq.work.InitObsSetting/MeasurementSpec` classes

I just tried this out and it works with InitObsSetting.

Reply:
Good idea.

It's mildly annoying that the interim reprs are both 1) very long and 2) going to change. Especially if we're codifying them in the json test data (which is necessary because the repr files are used to construct the example objects)

Reply:
Agreed - this is the complexity cost of "work in progress" features in this case. Still less than a feature branch + constant merging in my opinion - but that's another alternative.

Reply:
did the above